lib/koans.ex

```diff
@@ -82,11 +82,32 @@ defmodule Koans do
   end
 
   defmacro __before_compile__(env) do
-    koans = Module.get_attribute(env.module, :koans) |> Enum.reverse
+    koans = koans(env)
+    intro = extract_into(env)
     quote do
       def all_koans do
         unquote(koans)
       end
+
+      def intro do
+        unquote(intro)
```

Comment:
It seems like the culmination of this would be that all koans have intros and the metaprogramming is no longer needed, so I like the idea of including empty @intro to avoid the metaprogramming.